src/app/core/identity-provider/auth0.identity-provider.ts

```diff
@@ -92,65 +92,13 @@ export class Auth0IdentityProvider implements IdentityProvider {
           )
         ),
         whenTruthy(),
-        switchMap(idToken =>
-          this.apiService
-            .post<UserData>('users/processtoken', {
-              id_token: idToken,
-              options: ['CREATE_USER'],
-            })
-            .pipe(
-              tap(() => {
-                this.store.dispatch(loadUserByAPIToken());
-              }),
-              switchMap((userData: UserData) =>
-                combineLatest([
-                  this.store.pipe(select(getLoggedInCustomer)),
-                  this.store.pipe(select(getUserLoading)),
-                ]).pipe(
-                  filter(([, loading]) => !loading),
-                  first(),
-                  switchMap(([customer]) =>
-                    iif(
-                      () => !customer,
-                      this.router.navigate(['/register', 'sso'], {
-                        queryParams: {
-                          userid: userData.businessPartnerNo,
-                          firstName: userData.firstName,
-                          lastName: userData.lastName,
-                        },
-                      }),
-                      of(false)
-                    )
-                  ),
-                  switchMap((navigated: boolean) =>
-                    navigated || navigated === null
-                      ? race(
-                          this.store.pipe(
-                            select(getSsoRegistrationRegistered),
-                            whenTruthy(),
-                            tap(() => {
-                              this.store.dispatch(loadUserByAPIToken());
-                            })
-                          ),
-                          this.store.pipe(
-                            select(getSsoRegistrationCancelled),
-                            whenTruthy(),
-                            mapTo(false),
-                            tap(() => this.router.navigateByUrl('/logout'))
-                          )
-                        )
-                      : of(navigated)
-                  )
-                )
-              ),
-              switchMapTo(this.store.pipe(select(getUserAuthorized), whenTruthy(), first())),
-              catchError((error: HttpError) => {
-                this.apiTokenService.removeApiToken();
-                this.triggerLogout();
-                return of(error);
-              })
-            )
-        )
+        switchMap(idToken => {
+          const inviteUserId = window.sessionStorage.getItem('invite-userid');
+          const inviteHash = window.sessionStorage.getItem('invite-hash');
+          return inviteUserId
```

Comment:
Shouldn't you check that both values are truthy? You're passing both values to the `inviteRegistration()` function.